src/binary-reader.h

```diff
@@ -59,6 +59,11 @@ struct TypeMut {
 };
 using TypeMutVector = std::vector<TypeMut>;
 
+struct AlignMem {
+  Address alignment_log2;
+  Index memidx;
```

Comment:
Is there some reason in particular to wrap these two into a struct?   Do we do this for other parts of the binary reader API?   If we are going to do this why not also include the offset?  

Since this is makes the change so large maybe it could be split out into an NFC and landed first before any simd changes/fixes?

Reply:
Now that I think I've found all the locations where this is needed, I think we could split it back out and have every callsite go back to being `Index memidx, Address alignment_log2, Address offset`. Let me try that...

Reply:
Done. Should make for a somewhat cleaner diff.